src/dmd/backend/dwarfdbginf.d

```diff
@@ -612,47 +612,29 @@ static if (1)
      */
     void writeDebugFrameHeader(Outbuffer *buf)
     {
-        static struct DebugFrameHeader
+        buf.write32(I64 ? 20 : 16);             // length
+        buf.write32(0xFFFFFFFF);                // CIE_id
+        buf.writeByte(1);                       // version
+        buf.writeByte(0);                       // (no) augmentation
+        buf.writeuLEB128(1);                    // code alignment factor
+        buf.writesLEB128(0x80 - OFFSET_FAC);    // data alignment factor (-4 / -8)
```

Comment:
This is much slower than writing it as a struct. Why is this refactor needed?

Reply:
This is needed, because the struct won't be emitted entirely.
Please see [DWARF5.pdf](http://www.dwarfstd.org/doc/DWARF5.pdf).
- Chapter 7.4
> In the 32-bit DWARF format, an initial length field (see Section 7.2.2 on page
> 184) is an unsigned 4-byte integer (which must be less than 0xfffffff0 ); in
> the 64-bit DWARF format, an initial length field is 12 bytes in size, and has
> two parts:
> • The first four bytes have the value 0xffffffff .
> • The following eight bytes contain the actual length represented as an
> unsigned 8-byte integer.

Which implies a condition.

- Chapter 6.4, page 175
> 5. address_size (ubyte)
> ...
> 6. segment_selector_size (ubyte)

These two added fields also imply a condition (only emitted for DWARF v4 and v5).

I just saw that `Outbuf` has a `writen`, `writeByten`, ... delegates that don't reserve memory before writing, I'll use these one instead, but if you have a better solution, I'll gladly take it.

Reply:
To deal with conditions, do what I've done elsewhere - have two structs, and one conditional that picks which one.

I'm told that people are building D projects that take 15 minutes to compile. Having all these slowdowns bit by bit, like barnacles on a ship, slow it down.